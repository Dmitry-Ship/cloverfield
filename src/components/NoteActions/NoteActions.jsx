import PropTypes from 'prop-types';
import React from 'react';
import ColorMenu from '../ColorMenu';
import NoteFileUploader from '../NoteFileUploader';
import { icon } from './NoteActions.scss';

const NoteActions = ({ color, onSetColor, onChange, id, children, className }) => {
  const style = {
    display: 'flex',
    flexWrap: 'nowrap',
    justifyContent: 'space-around',
    alignItems: 'center',
  };

  return (
    <div style={style} className={className}>
      <ColorMenu color={color} position="topRight" onSetColor={onSetColor} className={icon} />
      <NoteFileUploader onChange={onChange} className={icon} id={id} />
      {children}
    </div>
  );
};

NoteActions.propTypes = {
  color: PropTypes.string,
  id: PropTypes.string.isRequired,
  onSetColor: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.any,
};

export default NoteActions;
