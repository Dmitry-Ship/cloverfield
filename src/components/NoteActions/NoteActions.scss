.icon {
  cursor: pointer;
  color: rgba(black, 0.3);
  transition: all 150ms ease-in-out;
  &:hover {
    color: black;
  }
}
