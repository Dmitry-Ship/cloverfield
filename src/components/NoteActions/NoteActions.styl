.icon
  cursor pointer
  color rgba(black, 0.3)
  // opacity 0
  transition all 150ms ease-in-out
  &:hover
    color black
