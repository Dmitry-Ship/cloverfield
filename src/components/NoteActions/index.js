export { default } from './NoteActions.jsx';
