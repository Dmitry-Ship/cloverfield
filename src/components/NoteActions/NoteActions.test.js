import React from 'react';
import renderer from 'react-test-renderer';
import NoteActions from './NoteActions';

describe('<NoteActions />', () => {
  const onSetColor = jest.fn();
  const onChange = jest.fn();
  

  it('should render correctly', () => {
    const tree = renderer.create(
      <NoteActions id="1" onSetColor={onSetColor} onChange={onChange} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children when position passed in', () => {
    const tree = renderer.create(
      <NoteActions id="1" onSetColor={onSetColor} onChange={onChange} >Hello World</NoteActions>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
