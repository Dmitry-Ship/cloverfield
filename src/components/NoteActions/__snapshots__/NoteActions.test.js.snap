// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteActions /> renders children when position passed in 1`] = `
<div
  className={undefined}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexWrap": "nowrap",
      "justifyContent": "space-around",
    }
  }
>
  <i
    className="colorMenu icon material-icons"
  >
    palette
    <div
      className="topRight popupMenu"
    >
      <div
        className="palette"
      >
        <span
          className="white chosen"
          onClick={[Function]}
        />
        <span
          className="red "
          onClick={[Function]}
        />
        <span
          className="orange "
          onClick={[Function]}
        />
        <span
          className="yellow "
          onClick={[Function]}
        />
        <span
          className="grey "
          onClick={[Function]}
        />
        <span
          className="blue "
          onClick={[Function]}
        />
        <span
          className="teal "
          onClick={[Function]}
        />
        <span
          className="green "
          onClick={[Function]}
        />
      </div>
    </div>
  </i>
  <div>
    <label
      className="icon label"
      htmlFor="1"
    >
      <i
        className="icon material-icons"
      >
        image
      </i>
    </label>
    <input
      accept="image/*"
      id="1"
      multiple={true}
      name="image"
      onChange={[Function]}
      required={false}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
  Hello World
</div>
`;

exports[`<NoteActions /> should render correctly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexWrap": "nowrap",
      "justifyContent": "space-around",
    }
  }
>
  <i
    className="colorMenu icon material-icons"
  >
    palette
    <div
      className="topRight popupMenu"
    >
      <div
        className="palette"
      >
        <span
          className="white chosen"
          onClick={[Function]}
        />
        <span
          className="red "
          onClick={[Function]}
        />
        <span
          className="orange "
          onClick={[Function]}
        />
        <span
          className="yellow "
          onClick={[Function]}
        />
        <span
          className="grey "
          onClick={[Function]}
        />
        <span
          className="blue "
          onClick={[Function]}
        />
        <span
          className="teal "
          onClick={[Function]}
        />
        <span
          className="green "
          onClick={[Function]}
        />
      </div>
    </div>
  </i>
  <div>
    <label
      className="icon label"
      htmlFor="1"
    >
      <i
        className="icon material-icons"
      >
        image
      </i>
    </label>
    <input
      accept="image/*"
      id="1"
      multiple={true}
      name="image"
      onChange={[Function]}
      required={false}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
</div>
`;
